web/dashboard/src/views/footer/footer.js

```diff
@@ -0,0 +1,96 @@
+import { Component } from 'react';
+import { formatDistance } from "date-fns";
+
+export default class Footer extends Component {
+    constructor(props) {
+        super(props);
+        let reduxState = this.props.store.getState();
+
+        this.state = {
+            gitopsCommits: reduxState.gitopsCommits,
+            envs: reduxState.envs
+        };
+        this.props.store.subscribe(() => {
+            let reduxState = this.props.store.getState();
+
+            this.setState({
+                gitopsCommits: reduxState.gitopsCommits,
+                envs: reduxState.envs
+            });
+        });
+    }
+
+    renderGitopsCommit(gitopsCommit) {
+        if (gitopsCommit === undefined) {
+            return null
+        }
+
+        const dateLabel = formatDistance(gitopsCommit.created * 1000, new Date());
+
+        const color = gitopsCommit.status.includes("Succeeded") ?
```

Comment:
Can you iterate over this nested ternary?
The goal should be: don't make me think when I look at it.